src/hbb/processors/categorizer.py

```diff
@@ -0,0 +1,329 @@
+            del trigger
+        else:
+            selection.add('trigger', ak.values_astype(ak.ones_like(events.run), bool))
+
+        if isRealData:
+            selection.add('lumimask', lumiMasks[self._year[:4]](events.run, events.luminosityBlock))
+        else:
+            selection.add('lumimask', ak.values_astype(ak.ones_like(events.run), bool))
+
+        if isRealData:
+            trigger = ak.values_astype(ak.zeros_like(events.run), bool)
+            for t in self._muontriggers[self._year]:
+                if t in events.HLT.fields:
+                    trigger = trigger | events.HLT[t]
+            selection.add('muontrigger', trigger)
+            del trigger
+        else:
+            selection.add('muontrigger', ak.values_astype(ak.ones_like(events.run), bool))
+
+        metfilter = ak.values_astype(ak.ones_like(events.run), bool)
+        for flag in self._met_filters[self._year]['data' if isRealData else 'mc']:
+            metfilter = dask.array.bitwise_and(metfilter, events.Flag[flag])
+        selection.add('metfilter', metfilter)
+        del metfilter
+
+        fatjets = events.FatJet
+        fatjets['msdcorr'] = fatjets.msoftdrop
+        fatjets['qcdrho'] = 2 * np.log(fatjets.msdcorr / fatjets.pt)
+
+        selection.add('2FJ', ak.num(fatjets, axis=1) == 2)
+        selection.add('not2FJ', ak.num(fatjets, axis=1) != 2)
+
+        candidatejet = fatjets[
+            (fatjets.pt > 200)
+            & (abs(fatjets.eta) < 2.5)
+            & fatjets.isTight 
+        ]
+
+        candidatejet = candidatejet[:, :2]
+        candidatejet = ak.firsts(candidatejet[ak.argmax(candidatejet.particleNet_XbbVsQCD, axis=1, keepdims=True)])
+
+        bvl = candidatejet.particleNet_XbbVsQCD
+
+        selection.add('minjetkin',
+            (candidatejet.pt >= 450)
+            & (candidatejet.pt < 1200)
+            & (candidatejet.msdcorr >= 40.)
+            & (candidatejet.msdcorr < 201.)
+            & (abs(candidatejet.eta) < 2.5)
+        )
+        selection.add('minjetkinmu',
+            (candidatejet.pt >= 400)
+            & (candidatejet.pt < 1200)
+            & (candidatejet.msdcorr >= 40.)
+            & (candidatejet.msdcorr < 201.)
+            & (abs(candidatejet.eta) < 2.5)
+        )
+        selection.add('jetid', candidatejet.isTight)
+        selection.add('ddbpass', (bvl >= 0.5))
+
+        jets = events.Jet
+        jets = jets[
+            (jets.pt > 30.)
+            & (abs(jets.eta) < 5.0)
+            & jets.isTight
+        ]
+
+        # only consider first 4 jets to be consistent with old framework
+        jets = jets[:, :4]
+        dphi = abs(jets.delta_phi(candidatejet))
+        selection.add('antiak4btagMediumOppHem', ak.max(jets[dphi > np.pi / 2].btagDeepFlavB, axis=1, mask_identity=False) < 0.3086) 
+        ak4_away = jets[dphi > 0.8]
+        selection.add('ak4btagMedium08', ak.max(ak4_away.btagDeepFlavB, axis=1, mask_identity=False) > 0.3086) 
```

Comment:
the working point (0.3086) will vary depending on the tagger and the year, suggest to put this in a dictionary that could be loaded from the json